### PR TITLE
Port changes from dotnet/runtimelab

### PR DESCRIPTION
**Pull request by jkotas**
- Fix build errors in System.Globalization.Native with libraries/Native warning level, add System.Globalization.Native back to the libraries/Native build to protect it,
- Misc other changes

### CODE REVIEW

**Review by MichalStrehovsky — APPROVED** [review]
Looks good modulo the `PInvokeILStubMethodIL` reference that will have to go under an ifdef. Thank you!

File: src/libraries/Native/Unix/System.Globalization.Native/pal_collation.c [hunk 1]
```diff
@@ -568,15 +568,17 @@ static const int32_t GetSearchIteratorUsingCollator(
     usearch_setText(*pSearchIterator, lpSource, cwSourceLength, &err);
     if (!U_SUCCESS(err))
     {
-        int32_t r = RestoreSearchHandle(pSortHandle, *pSearchIterator, options);
+        int32_t r;
+        r = RestoreSearchHandle(pSortHandle, *pSearchIterator, options);
```
Comment (am11) [inline src/libraries/Native/Unix/System.Globalization.Native/pal_collation.c:572]: Just curious, was it causing an error due to assignment and declaration on the same line?

Comment (jkotas) [reply src/libraries/Native/Unix/System.Globalization.Native/pal_collation.c:572]: It produce unused variable warning/error in release builds.

Comment (jkotas) [reply src/libraries/Native/Unix/System.Globalization.Native/pal_collation.c:572]: Another way to "fix this" would be to add `(void)r;`. The libraries shims use both patterns.

Comment (am11) [reply src/libraries/Native/Unix/System.Globalization.Native/pal_collation.c:572]: Ah makes sense. Another `(void)` pattern we used in runtime repo is dropping the variable altogether, e.g.:

```c++
(void)RestoreSearchHandle(pSortHandle, *pSearchIterator, options);
```

Comment (jkotas) [reply src/libraries/Native/Unix/System.Globalization.Native/pal_collation.c:572]: This works only if it is truly unused. It is used here for debug-only assert.

**Pull request merged**